R/ggplotMDS.R

```diff
@@ -357,26 +577,45 @@ MDS_var_explained <- function(mds,
                               topN = 10,
                               cumVarLimit = 0.9,
                               barColor="dodgerblue4",
-                              barFill = "dodgerblue3",
-                              barWidth = 0.65,
-                              barSize = 0.1) {
+                              barWidth = 0.65) {
+    plotType <- tolower(plotType)
 
     assertthat::assert_that(!missing(mds),
                             msg = "mds is required and must be specified.")
-    assertthat::assert_that(plotType %in% c("canvasXpress", "ggplot"),
+    assertthat::assert_that(!is.null(plotType),
+                            plotType %in% c("canvasxpress", "ggplot"),
                             msg = "Plot type must be either canvasXpress or ggplot.")
 
+    #input parameter validation
+    if (any(is.null(topN), length(topN) != 1, !is.numeric(topN), topN < 1)) {
+        warning("topN should be a numeric value. Assigning default value 10.")
+        topN <- 10
+    }
+
+    if (any(is.null(cumVarLimit), length(cumVarLimit) != 1, !is.numeric(cumVarLimit), cumVarLimit < 0, cumVarLimit > 1)) {
+        warning("cumVarLimit should be a single numeric value between 0 and 1. Assigning default value 0.9.")
+        cumVarLimit <- 0.9
+    }
+
+    if (any(is.null(barColor), !is.character(barColor), length(barColor) != 1, .rgbaConversion(barColor) == "invalid value")) {
+        warning("barColor specified is not valid. Assigning default value 'dodgerblue4'.")
+        barColor <- "dodgerblue4"
+    }
+
+    if (any(is.null(barWidth), !is.numeric(barWidth), length(barWidth) != 1, barWidth < 0, barWidth > 1)) {
+        warning("barWidth should be a single numeric value between 0 and 1. Assigning default value 0.65.")
+        barWidth <- 0.65
+    }
+
     if (!("MDS" %in% class(mds))) {
         mds <- limma::plotMDS(mds, plot = FALSE)
     }
 
     mds.distances <- mds %$% distance.matrix %>% as.dist
-
     mdsvals <- mds.distances %>%
         {suppressWarnings(cmdscale(., k = ncol(mds$distance.matrix) - 1))} %>%
         magrittr::set_colnames(stringr::str_c("Dim", seq_len(ncol(.)))) %>%
```

Comment:
add **magrittr::set_colnames** to **importFrom**

Reply:
ad **stringr::str_c** to **importFrom**

Reply:
For ggplot2 and magrittr -I am seeting the entire package name being included in the import section. Seeing this in multiple functions. Will check with Connie before I make the change.